c++/include/orc/Vector.hh

```diff
@@ -113,6 +115,35 @@ namespace orc {
     DataBuffer<int64_t> length;
   };
 
+  struct StringDictionary {
+    StringDictionary(MemoryPool& pool);
+    DataBuffer<char> dictionaryBlob;
+
+    // Offset for each dictionary key entry.
+    DataBuffer<int64_t> dictionaryOffset;
```

Comment:
uint32_t? I assume 4G of dictionary size is more than enough.

Reply:
Right now the RleDecoder only support int64_t*